src/modules/include-exclude/includes/Plugin_Enabled.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * This file contains the core plugin functions.
+ *
+ * @package Wordlift
+ */
+
+namespace WordLift\Modules\Include_Exclude;
+
+/**
+ * Define the JSON-LD Enabled class.
+ *
+ * The Core class instance will hook the `wl_jsonld_enabled` filter.
+ *
+ * @package Wordlift
+ */
+class Plugin_Enabled {
+
+	/**
+	 * Register hooks.
+	 */
+	public function register_hooks() {
+		add_filter( 'wl_is_enabled', array( $this, 'wl_is_enabled' ) );
+		add_filter( 'wl_jsonld_term_html_output', array( $this, 'may_be_exclude_nitropack' ) );
+		add_filter( 'wl_jsonld_post_html_output', array( $this, 'may_be_exclude_nitropack' ) );
+	}
+
+	/**
+	 * Modify JSON-LD if Nitropack active.
+	 *
+	 * @param $html
+	 *
+	 * @return array|mixed|string|string[]
+	 */
+	public function may_be_exclude_nitropack( $html ) {
+		if ( ! defined( 'NITROPACK_VERSION' ) ) {
+			return $html;
+		}
+		$html = str_replace( 'id="wl-jsonld"', 'id="wl-jsonld" nitro-exclude', $html );
+		$html = str_replace( 'id="wl-jsonld-term"', 'id="wl-jsonld-term" nitro-exclude', $html );
+
+		return $html;
+	}
+
+	/**
+	 * Enable/Disable WordLift Plugin.
+	 *
+	 * @param $enabled
+	 *
+	 * @return bool|mixed
+	 */
+	public function wl_is_enabled( $enabled ) {
+
+		// Always enable wordlift on admin and rest api pages.
+		if ( is_admin() || $this->is_rest_request() ) {
+			return $enabled;
+		}
+
+		$path        = strtok( (string) $_SERVER['REQUEST_URI'], '?' ); // phpcs:ignore
+		$options     = get_option( 'wl_exclude_include_urls_settings' );
+		$current_url = trailingslashit( home_url( $path ) );
+
+		// Bail out if URLs are not set.
+		if ( empty( $options['urls'] ) ) {
+			return $enabled;
+		}
+
+		// Set a default state
+		$default_state = $options['include_exclude'] === 'exclude' ? true : false;
+
+		// Get URLs into an array from settings, trim them and make absolute if needed.
+		$urls = array_map(
+			function ( $url ) {
+				$url = trim( $url );
+				if ( substr( $url, 0, 4 ) !== 'http' ) {
+					return trailingslashit( home_url( $url ) );
+				}
+
+				// Add a trailing slash and return the url
+				return trailingslashit( $url );
+			},
+			explode( PHP_EOL, $options['urls'] )
+		);
+
+		foreach ( $urls as $url ) {
+			if ( $url === $current_url ) {
+				return ! $default_state;
+			}
+		}
+		return $default_state;
```

Comment:
What's this, surely there's a better and more efficient way to check if a URL is in the options by using a regex.